fahrschein/src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -341,6 +342,7 @@ void runInternal() throws IOException, BackoffException {
                 throw e;
             }
         }
+        jsonInput.close();
```

Comment:
I see your point and agree with you, exceptions should not be used for flow control.

But I see the point that the logic gets more and more complicated. I will introduce a custom exception for now, maybe in the future a better way of dealing with it is discovered.